### PR TITLE
build: update mud to next14

### PR DESCRIPTION
**Pull request by noyyyy**

### CODE REVIEW

File: packages/contracts/test/AutoBattleTest.t.sol [hunk 1]
```diff
@@ -13,36 +13,39 @@ import {GameStatus} from "src/codegen/common.sol";
...
 import {TestCommon} from "./TestCommon.t.sol";
+import {StoreSwitch} from "@latticexyz/store/src/StoreSwitch.sol";
 
 contract AutoBattleSystemTest is MudTest {
     IWorld public world;
 
     function setUp() public override {
         super.setUp();
         world = IWorld(worldAddress);
+        StoreSwitch.setStoreAddress(worldAddress);
```
Comment (ClaudeZsb) [inline packages/contracts/test/AutoBattleTest.t.sol]:
```solidity
function setUp() public override {
    super.setUp();
    TestCommon.normalStart(vm, world);
}
```
`IWorld public world` is not needed anymore and `StoreSwitch` has been set within `setUp()` of `MudTest`.

Comment (noyyyy) [reply packages/contracts/test/AutoBattleTest.t.sol]: No. You need to call the the world and `IWorld` provide the right interface.

Comment (noyyyy) [reply packages/contracts/test/AutoBattleTest.t.sol]: I can delete the line of `StoreSwitch`

Comment (ClaudeZsb) [reply packages/contracts/test/AutoBattleTest.t.sol]: > No. You need to call the the world and `IWorld` provide the right interface.

you're right. Just forget that.

**Review by ClaudeZsb — APPROVED** [review]

**Pull request merged**